### ISSUE TITLE
Add a variable for blacklisting linters

### ISSUE THREAD

**Issue by kbobyrev** [w0rp/ale#1453]
Project README FAQ [suggests](https://github.com/w0rp/ale#5i-how-do-i-disable-particular-linters) disabling linters via establishing a list of enabled linters like this:

```
let g:ale_linters = {
\   'javascript': ['eslint'],
\}
```

While this looks to be a feasible solution, I would rather like to have something `ale_linters_blacklist` list filled with those I don't want to use. Otherwise, I would have to pull every single linter from the project documentation (which itself is very time-consuming) and only then remove unwanted ones.

Also, I would be happy to have an option to completely disable all checks for a specific language. I'm mostly writing C++ code and ale slows Vim down eating tons of RAM whenever a compile error happens even in very small toy projects. Therefore I'd be happy if I could just disable linters for C++, otherwise editor becomes unresponsive and almost unusable at times.

Overall, I'd be happy to see any other options for filtering out unwanted linters.

**Comment by w0rp** [comment]
You have the following options available.

1. Disable linters for a filetype by setting a filetype key in `g:ale_linters` to an empty list. `[]`.
2. Use `let g:ale_linters_explicit = 1` so no linters will be enabled for anything apart from the ones you specify yourself.
3. Use `b:ale_linters` in an ftplugin file, or anything you can run for each buffer, to configure linters for that filetype.

See`:help g:ale_linters` and `:help g:ale_linters_explicit` for more information. There wouldn't be much point in naming linters you don't want to run with a `g:ale_linters_blacklist` variable, because you'd only be one `git pull` away from accidentally running another linter someone added. It's better to name only the linters you want.

**Comment by omtcvxyz** [comment]
Hi @w0rp! Thank you for describing these options. While I can get your point I still have a concern:

> because you'd only be one git pull away from accidentally running another linter someone added

This is indeed correct. However, Identifying a bad linter is always an easy task: "Hey, this linter slows down my editor" / "Hey, this linter is not very useful and takes up a lot of RAM". Therefore, whenever someone adds a linter I would certainly understand if it's affecting my performance, therefore it's pretty easy to disable it given an opportunity to blacklist certain linters.

However, if I only whitelist them via `g:ale_linters` I wouldn't know that someone added a new-useful-linter-9000 unless I frequently check it. That would be a problem.

Long story short: I think it's easy to understand that a linter is not wanted over the course of its usage and blacklist it, but it would be way harder if I actually have to go to the docs and check whether there is anything useful I'd want to pull into my workflow.

**Comment by w0rp** [comment]
I see your point. I'll add an enhancement tag to this issue, and someone can pick it up. I might pick it up eventually. I have some recommendations.

1. Name the variable `ale_linters_blacklist`, and implement global and buffer-local variables.
2. Have the buffer variable override the global variable completely for that buffer. (Linters can be removed from a global blacklist this way.)
3. For both scopes of variable, support a `List` of linter names, for filtering out linters to run after they have been selected with `g:ale_linters` or `b:ale_linters`, and also a `Dictionary` mapping filetypes to linters to ignore for those filetypes.
4. Linters named in the blacklist variable should exclude those named in `g:ale_linters` or `b:ale_linters` without throwing any exceptions.

**Comment by omtcvxyz** [comment]
Great, thanks! You can also add "help wanted" or any other beginner-friendly (e.g. "good-first-issue") tag since I think it would be relatively easy to implement this feature, I'd do that myself but I, unfortunately, don't have time right now.

**Comment by hadriann** [comment]
A bit off-topic, but is there a way to enable / run a javascript linter (eslint) on html files? Thanks.

**Comment by w0rp** [comment]
Yes. What you need to do is install an HTML plugin for ESLint, or whatever it needs to check HTML files in your terminal, and then you need to set up aliases so you can run JavaScript linters for HTML files. You can select `'eslint'` as a linter for HTML filetypes. See this FAQ entry: https://github.com/w0rp/ale#5xii-how-can-i-check-jsx-files-with-both-stylelint-and-eslint

You probably need something like this.

```vim
" Use eslint only to check HTML files
let g:ale_linters = {'html': ['eslint']} 
" Run both HTML and JavaScript linters for HTML files
let g:ale_linter_aliases = {'html': ['html', 'javascript']}
```

See the documentation for each option.

**Comment by w0rp** [comment]
Now I also want to be able to run LSP tools while ignoring diagnostics for them. I might design this option so it can be used for both purposes, and I might name it `ale_linters_ignore` instead.

**Comment by w0rp** [comment]
Now linter results can be ignored with `g:ale_linters_ignore`.
